File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numsDict = new HashMap<>();
        for (int i = 0; i < nums.length; i++ ) {
            if (numsDict.get(target - nums[i]) != null) {
                return new int[] {numsDict.get(target - nums[i]), i};
            }
            else {
                numsDict.put(nums[i], i);
            }
        }
        return null;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-3.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        # Intialize the hashmap to check the numbers
        num_dict = {}

        for i in nums:
            if i in num_dict:
                return True
            else:
                num_dict[i] = 1
        return False
        